### DISCUSSION TITLE
Editing plaintext assets

### DISCUSSION THREAD

**Discussion by devurandom** [sveltia/sveltia-cms#67] [Q&A] [answered]
I am using https://11ty.dev and extended it to support generating SVGs from a plaintext file. This only works well when I place the source file under `assets/`, so that is what I did: `assets/plain.txt => _site/assets/plain.svg`. Since the source file is plaintext, I expected to be able to edit it using Sveltia CMS, but it is not even displayed in the UI clicking on the assets icon, which takes me to `/admin/#/assets`.

How can I edit plaintext files, that are located under `assets/` in my source repository, with Sveltia CMS?

**Comment by kyoshino** [comment]
- I’ll make sure plaintext files will be displayed in the asset library. It’s just a bug, I guess.
- I’m planning to add a simple image editor to the asset library (cropping, resizing, etc.) but wasn’t thinking about plaintext files. Maybe it’s not difficult to add a simple text edtior. Will check!

**Reply by kyoshino** [reply, marked as answer]
I just tested with the [11ty starter template](https://github.com/danurbanowicz/eleventy-sveltia-cms-starter) and looks like plaintext files appear in the asset library if configured properly. What I’ve changed:

Specify the assets folder in `.eleventy.js`:

```js
eleventyConfig.addPassthroughCopy("assets/"); 
```

And change the `media_folder` in `admin/config.yml`:

```yaml
media_folder: "assets"
```

<img width="505" alt="image" src="https://github.com/sveltia/sveltia-cms/assets/2929505/d65a2a04-4aa6-4fce-a10f-c6cbfc74ab40">

**Reply by devurandom** [reply]
Thanks for the hint! Moving the file to `assets/img/` made it appear in Sveltia! Now the only thing missing is showing a text input box. :) If I have to add a value to an array of file extensions to show such input field for, that would work for me.

**Reply by devurandom** [reply]
As a workaround I added the text->SVG transformer as a Markdown "highlighter", which allows me to embed the text into a "page": https://codeberg.org/devurandom/eleventy-plugin-text-mapper/src/branch/main/index.js#L17-L26
This is following the example of another 11ty plugin that renders PlantUML in this way: https://github.com/awaragi/eleventy-plugin-plantuml

**Comment by kyoshino** [comment]
Finally! Just shipped [v0.17.0](https://github.com/sveltia/sveltia-cms/releases/tag/v0.17.0) with a plaintext editor 📝 

<img width="505" alt="image" src="https://github.com/sveltia/sveltia-cms/assets/2929505/b13e4e0b-d9b1-4ead-bf37-0e9d29805b67">
